plenum/common/util.py

```diff
@@ -40,27 +40,32 @@
 Seconds = TypeVar("Seconds", int, float)
 
 
-def randomString(size: int = 20,
-                 chars = string.ascii_letters + string.digits) -> str:
+def randomString(size: int = 20) -> str:
     """
-    Generate a random string of the specified size
+    Generate a random string of the specified size,
+    DONOT use python provided random class its a Pseudo Random Number Generator
+    and not secure enough for our needs
 
     :param size: size of the random string to generate
-    :param chars: the set of characters to use to generate the random string. Uses alphanumerics by default.
     :return: the random string generated
     """
 
-    if not hasattr(chars, "__getitem__"):
-        # choice does not work with non indexed containers
-        chars = list(chars)
+    def randomStr(size):
+        assert (size > 0), "Expected random string size cannot be less than 1"
+        rstr = randombytes(size).hex()
```

Comment:
I will test this for performance tonight and see how this performs. I suspect the 2 calls to random in case of odd size is probably going to slow it down